From 65d55adea4d8499e9f99b8e45f52089d19df502f Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Wed, 22 Jan 2025 14:32:03 +0100
Subject: [PATCH] Revert "applets/kickoff: change category at most once per
 frame"

This reverts commit d5b9b7991a7e262bc34bf46befb56cfc4f029f51.
The functionality is so broken as I question if this was even
tested at all. This breaks changing the index with keys as
currentIndex is always reset to pendingIndex and for the
same reason by clicking. Resulting in being unable to
change categories at all. Even after fixing this the design
doesn't work, as not changing index doesn't cause a redraw
which means no frameSwapped occurs.
Further the CI pipeline failing on kickofftest was ignored
when merging.
---
 .../contents/ui/AbstractKickoffItemDelegate.qml        | 10 +++-------
 .../kickoff/package/contents/ui/KickoffListView.qml    |  9 ---------
 2 files changed, 3 insertions(+), 16 deletions(-)

diff --git a/applets/kickoff/package/contents/ui/AbstractKickoffItemDelegate.qml b/applets/kickoff/package/contents/ui/AbstractKickoffItemDelegate.qml
index d8cc0d7a02..d350057272 100644
--- a/applets/kickoff/package/contents/ui/AbstractKickoffItemDelegate.qml
+++ b/applets/kickoff/package/contents/ui/AbstractKickoffItemDelegate.qml
@@ -182,13 +182,9 @@ T.ItemDelegate {
             if (!root.activeFocus) {
                 root.forceActiveFocus(Qt.MouseFocusReason)
             }
-            // If the view exports a custom property pending Index, use that one
-            // which will do event compression and ensure we don't do omre than a category switch per frame
-            if (root.view.hasOwnProperty("pendingIndex")) {
-                root.view.pendingIndex = index
-            } else {
-                root.view.currentIndex = index
-            }
+            // No need to check currentIndex first because it's
+            // built into QQuickListView::setCurrentIndex() already
+            root.view.currentIndex = index
         }
         onPressed: mouse => {
             // Select and focus on press to improve responsiveness and touch feedback
diff --git a/applets/kickoff/package/contents/ui/KickoffListView.qml b/applets/kickoff/package/contents/ui/KickoffListView.qml
index 5c03b954d2..6f18ee1963 100644
--- a/applets/kickoff/package/contents/ui/KickoffListView.qml
+++ b/applets/kickoff/package/contents/ui/KickoffListView.qml
@@ -11,7 +11,6 @@
 pragma ComponentBehavior: Bound
 
 import QtQuick
-import QtQuick.Window
 import QtQuick.Templates as T
 
 import org.kde.plasma.plasmoid
@@ -89,14 +88,6 @@ EmptyPage {
         property bool movedWithKeyboard: false
         property bool movedWithWheel: false
 
-        property int pendingIndex: 0
-        Connections {
-            target: root.Window.window
-            function onFrameSwapped() {
-                view.currentIndex = view.pendingIndex;
-            }
-        }
-
         Accessible.role: Accessible.List
 
         implicitWidth: {
